src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -171,114 +204,137 @@ private WordEmbeddingsTransform(IHost host, ModelLoadContext ctx, IDataView inpu
             else
             {
                 _modelKind = (PretrainedModelKind)ctx.Reader.ReadUInt32();
-                _modelFileNameWithPath = EnsureModelFile(Host, out _linesToSkip, (PretrainedModelKind)_modelKind);
+                _modelFileNameWithPath = EnsureModelFile(Host, out linesToSkip, (PretrainedModelKind)_modelKind);
             }
 
             Host.CheckNonWhiteSpace(_modelFileNameWithPath, nameof(_modelFileNameWithPath));
-            _currentVocab = GetVocabularyDictionary();
-            _outputType = new VectorType(NumberType.R4, 3 * _currentVocab.Dimension);
-            Metadata.Seal();
+            _currentVocab = GetVocabularyDictionary(linesToSkip);
+
+            InputType = new VectorType(PrimitiveType.FromKind(DataKind.Text));
+            OutputType = new VectorType(NumberType.R4, 3 * _currentVocab.Dimension);
         }
 
-        public static WordEmbeddingsTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+        public static WordEmbeddingsTransform Create(IHostEnvironment env, ModelLoadContext ctx)
         {
             Contracts.CheckValue(env, nameof(env));
             IHost h = env.Register(RegistrationName);
             h.CheckValue(ctx, nameof(ctx));
-            h.CheckValue(input, nameof(input));
             return h.Apply("Loading Model",
-                ch => new WordEmbeddingsTransform(h, ctx, input));
+                ch => new WordEmbeddingsTransform(h, ctx));
```

Comment:
>ch => new WordEmbeddingsTransform(h, ctx)) [](start = 16, length = 42)

Is this recursive  call?

Reply:
No, It does not seem to be.

---
In reply to: [217929300](https://github.com/dotnet/machinelearning/pull/928#discussion_r217929300) [](ancestors = 217929300)

Reply:
Seems like your comment is addressed, right?

---
In reply to: [218154567](https://github.com/dotnet/machinelearning/pull/928#discussion_r218154567) [](ancestors = 218154567,217929300)